vector/src/main/java/im/vector/notifications/NotificationUtils.kt

```diff
@@ -0,0 +1,845 @@
+     * @return the call notification.
+     */
+    @SuppressLint("NewApi")
+    fun buildPendingCallNotification(context: Context, roomName: String, roomId: String, matrixId: String, callId: String): Notification {
+        addNotificationChannels(context)
+
+        val builder = NotificationCompat.Builder(context, CALL_NOTIFICATION_CHANNEL_ID)
+                .setWhen(System.currentTimeMillis())
+                .setContentTitle(roomName)
+                .setContentText(context.getString(R.string.call_in_progress))
+                .setSmallIcon(R.drawable.incoming_call_notification_transparent)
+
+        // Display the pending call notification on the lock screen
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            builder.priority = NotificationCompat.PRIORITY_MAX
+        }
+
+        // Build the pending intent for when the notification is clicked
+        val roomIntent = Intent(context, VectorRoomActivity::class.java)
+                .putExtra(VectorRoomActivity.EXTRA_ROOM_ID, roomId)
+                .putExtra(VectorRoomActivity.EXTRA_MATRIX_ID, matrixId)
+                .putExtra(VectorRoomActivity.EXTRA_START_CALL_ID, callId)
+
+        // Recreate the back stack
+        val stackBuilder = TaskStackBuilder.create(context)
+                .addParentStack(VectorRoomActivity::class.java)
+                .addNextIntent(roomIntent)
+
+        // android 4.3 issue
+        // use a generator for the private requestCode.
+        // When using 0, the intent is not created/launched when the user taps on the notification.
+        //
+        val pendingIntent = stackBuilder.getPendingIntent(Random().nextInt(1000), PendingIntent.FLAG_UPDATE_CURRENT)
+
+        builder.setContentIntent(pendingIntent)
+
+        return builder.build()
+    }
+
+    /**
+     * Create a square bitmap from another one.
+     * It is centered.
+     *
+     * @param bitmap the bitmap to "square"
+     * @return the squared bitmap
+     * TODO Move to Bitmap util
+     */
+    private fun createSquareBitmap(bitmap: Bitmap?): Bitmap? {
+        var resizedBitmap: Bitmap? = null
+
+        if (null != bitmap) {
+            // convert the bitmap to a square bitmap
+            val width = bitmap.width
+            val height = bitmap.height
+
+            if (width == height) {
+                resizedBitmap = bitmap
+            } else if (width > height) {
+                resizedBitmap = Bitmap.createBitmap(
+                        bitmap,
+                        (width - height) / 2,
+                        0,
+                        height,
+                        height
+                )
+
+            } else {
+                resizedBitmap = Bitmap.createBitmap(
+                        bitmap,
+                        0,
+                        (height - width) / 2,
+                        width,
+                        width
+                )
+            }// higher than large
```

Comment:
Comments are now away from the code they explain

Reply:
Ok, fixed. Also this code has been moved to a dedicated class, so we can remove some code duplication :)